### ISSUE TITLE
Treeshaking issue in branch 'feature/031-all-module-types'

### ISSUE THREAD

**Issue by samarpanda** [TheLarkInn/webpack-workshop-2018#23]
Webpack 4 fundamentals
`git checkout feature/031-all-module-types`

My local environment
~ » node -v                                                                                                                     
v10.6.0
~ » npm -v                                                                                                                      
6.1.0

Index.js
```js
import nav from "./nav";
import { top, bottom } from "./footer";
import makeButton from "./button";
import { makeColorStyle } from "./button-styles";

console.log(
  nav(),
  top,
  bottom,
  makeButton("My first button!"),
  makeColorStyle("cyan")
);
```

button-styles.js
```js
const red = "color: red;";
const blue = "color: blue;";
const makeColorStyle = color => `color: ${color};`;

exports.red = red;
exports.blue = blue;
exports.makeColorStyle = makeColorStyle;
```

As per the workshop `npm run prod` should remove `red & blue` references from the build file as its not used. But its not happening.

```bash
~/learning/webpack-workshop-2018(feature/031-all-module-types) » npm run prod -- --display-optimization-bailout

> webpack-workshop-2018@1.0.0 prod /Users/apple/learning/webpack-workshop-2018
> npm run webpack -- --mode production "--display-optimization-bailout"


> webpack-workshop-2018@1.0.0 webpack /Users/apple/learning/webpack-workshop-2018
> webpack "--mode" "production" "--display-optimization-bailout"

Hash: 2e38741249cef67d5b18
Version: webpack 4.16.2
Time: 122ms
Built at: 07/27/2018 7:47:31 AM
  Asset      Size  Chunks             Chunk Names
main.js  1.18 KiB       0  [emitted]  main
[0] ./src/button.js 218 bytes {0} [built]
    ModuleConcatenation bailout: Module is not an ECMAScript module
[1] ./src/button-styles.js 190 bytes {0} [built]
    ModuleConcatenation bailout: Module is not an ECMAScript module
[2] ./src/index.js + 2 modules 359 bytes {0} [built]
    ModuleConcatenation bailout: Cannot concat with ./src/button-styles.js (<- Module is not an ECMAScript module)
    ModuleConcatenation bailout: Cannot concat with ./src/button.js (<- Module is not an ECMAScript module)
    | ./src/index.js 252 bytes [built]
    |     ModuleConcatenation bailout: Module is an entry point
    | ./src/nav.js 28 bytes [built]
    | ./src/footer.js 69 bytes [built]

```

main.js
```js
!function(e){var t={};function r(o){if(t[o])return t[o].exports;var n=t[o]={i:o,l:!1,exports:{}};return e[o].call(n.exports,n,n.exports,r),n.l=!0,n.exports}r.m=e,r.c=t,r.d=function(e,t,o){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:o})},r.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var o=Object.create(null);if(r.r(o),Object.defineProperty(o,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var n in e)r.d(o,n,function(t){return e[t]}.bind(null,n));return o},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,"a",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p="",r(r.s=2)}([function(e,t){e.exports=(e=>`Button: ${e}`)},function(e,t){t.red="color: red;",t.blue="color: blue;",t.makeColorStyle=(e=>`color: ${e};`)},function(e,t,r){"use strict";r.r(t);var o=r(0),n=r.n(o),u=r(1);console.log("nav","top","bottom",n()("My first button!"),Object(u.makeColorStyle)("cyan"))}]);
```

**Comment by samarpanda** [comment]
Is webpack tree-shaking works with ECMAScript modules i.e using 'export', but not with 'exports'?

**Comment by samarpanda** [comment]
I think this will fix it #24

**Comment by 1Marc** [comment]
I've merged this fix since it will give people expected output that matches the course, and hopefully pointing to this issue as an annotation will help people understand why the code in the branch is different to enable tree shaking.
